src/modules/CrashReporting.ts

```diff
@@ -16,9 +17,23 @@ export const setEnabled = (isEnabled: boolean) => {
 /**
  * Send handled JS error object
  * @param error Error object to be sent to Instabug's servers
+ * @param userAttributes (Optional) extra user attributes attached to the crash
+ * @param fingerprint (Optional) key used to customize how crashes are grouped together
+ * @param levelType different severity levels for errors
  */
-export const reportError = (error: ExtendedError) => {
-  return InstabugUtils.sendCrashReport(error, NativeCrashReporting.sendHandledJSCrash);
+export const reportError = (
+  error: ExtendedError,
+  userAttributes: Object | null,
+  fingerprint: string | null,
+  levelType: NonFatalErrorType,
+) => {
```

Comment:
We need to make `userAttributes`, `fingerprint`, `levelType` (I think we can rename it to `level`) optional, we should also turn them into an `options` object or something similar to named parameters in other languages for clarity as the function now has 4 parameters. Let's also turn `userAttributes` into `Record<string, string>` since `Object` would match too many unwanted types.

We will need to define an interface `NonFatalOptions` for the options, we can put it in `models` as we do with `InstabugConfig` and export it from `src/index.ts` to be accessible to users.

Here's how it might look.

```ts
interface NonFatalOptions {
  userAttributes?: Record<string, string>;
  fingerprint?: string;
  level?: NonFatalErrorLevel;
}
```

```suggestion
export const reportError = (
  error: ExtendedError,
  { userAttributes, fingerprint, level }: NonFatalOptions = {},
) => {
```